Lib/defcon/objects/font.py

```diff
@@ -714,15 +722,29 @@ def save(self, path=None, formatVersion=None, removeUnreferencedImages=False, pr
         # otherwise fallback to 3
         elif formatVersion is None:
             formatVersion = 3
-        # if down-converting, use a temp directory
-        convertinginPlace = False
-        if path == self._path and formatVersion != self._ufoFormatVersion:
-            convertinginPlace = True
+        # if down-converting in-place or "saving as" to a pre-existing path,
+        # we first write to a temporary folder, then move to destination
+        overwritePath = None
+        if ((not saveAs and formatVersion != self._ufoFormatVersion) or
+                (saveAs and os.path.exists(path))):
             saveAs = True
+            overwritePath = path
             path = os.path.join(tempfile.mkdtemp(), "temp.ufo")
         try:
             # make a UFOWriter
-            writer = UFOWriter(path, formatVersion=formatVersion)
+            try:
+                writer = UFOWriter(path, formatVersion=formatVersion)
+            except UFOLibError:
+                if overwritePath is None and os.path.exists(path):
+                    logger.exception("Invalid ufo found '%s', the existing ufo "
```

Comment:
this will print the "Invalid ufo found..." message as well as the exception traceback.